LibSWBF2/Chunks/HeaderNames.cpp

```diff
@@ -55,15 +67,29 @@ namespace LibSWBF2
 
 	Types::String ChunkHeader::ToString() const
 	{
+		auto soundLookup = KNOWN_SOUND_HEADERS.find(*this);
+		if (soundLookup != KNOWN_SOUND_HEADERS.end())
+		{
+			return soundLookup -> second.c_str();
+		}
+
 		std::string result;
-		result += m_Name[0];
-		result += m_Name[1];
-		result += m_Name[2];
-		result += m_Name[3];
+		if (!IsPrintableHeader(*this))
+		{
+			result = fmt::format("0x{0:x}", m_Magic);
+		}
+		else 
+		{
+			result += m_Name[0];
+			result += m_Name[1];
+			result += m_Name[2];
+			result += m_Name[3];
+		}
+
 		return result.c_str();
 	}
 
-	bool IsValidHeader(const ChunkHeader hedr)
+	bool IsPrintableHeader(const ChunkHeader hedr)
```

Comment:
Given that sound chunks are hashes, it no longer made sense to equate valid with four printable characters